### PR TITLE
added becomeSetHome method to add -H flag when using sudo

### PR DESCRIPTION
**Pull request by gabrielef**
I have added the becomeSetHome method to add the -H flag when become is used. In this way is possible to use home directory of the current user.

### CODE REVIEW

**Comment by gabrielef** [comment]
This feature add the -H flasg to the current implementation of the become() method:
`sudo -H -u <user>`

In this way the ~ directory will be the _/home/user_ directory. This could be useful because not in all systems is the default behaviour as reported in the sudo man:

> -H, --set-home
Request that the security policy set the HOME environment variable to the home directory specified by the target user's password database entry. Depending on the policy, this may be the default behavior.

Ansible is doing this as a default behaviour.

**Comment by antonmedv** [comment]
Maybe we can do it as default too and merge in into v7?

**Comment by gabrielef** [comment]
I think it could be good as a default behaviour too.

My current workaround, using version 6.x is to declare become as below:

`->become('{{become_user}} -H ')`

File: src/Host/Host.php [hunk 1]
```diff
@@ -209,31 +210,49 @@ public function multiplexing(bool $multiplexing = true)
...
+    /**
+     * @return bool
+     */
+    public function isBecomeSetHome()
+    {
+        return $this->becomeSetHome;
+    }
+
+    /**
+     * @param bool $becomeSetHome
```
Comment (staabm) [inline src/Host/Host.php]: would be great to have more docs on this "setter", so people know why/when this switch will be usefull.

**Comment by antonmedv** [comment]
Let's use -H by default and merge it then.

**Pull request merged**